WordPress/Classes/ViewRelated/Post/AuthorFilterViewController.swift

```diff
@@ -0,0 +1,224 @@
+import UIKit
+import Gridicons
+
+/// Displays a simple table view picker list to choose between author filters
+/// for a post.
+///
+class AuthorFilterViewController: UITableViewController {
+    private typealias AuthorFilter = PostListFilterSettings.AuthorFilter
+
+    /// An optional gravatar email address. If provided, this will be used to
+    /// display a gravatar icon alongside the "Only Me" posts
+    var gravatarEmail: String? = nil
+
+    /// The currently selected author filer
+    var currentSelection: PostListFilterSettings.AuthorFilter
+
+    /// An optional block, which will be called whenever the user selects an
+    /// item in the filter list.
+    var onSelectionChanged: ((PostListFilterSettings.AuthorFilter) -> Void)? = nil
+
+    private var selectedIndexPath: IndexPath? {
+        guard let row = rows.index(of: currentSelection) else {
+            return nil
+        }
+
+        return IndexPath(row: row, section: 0)
+    }
+
+    private let rows = [
+        AuthorFilter.mine,
+        AuthorFilter.everyone
+    ]
+
+    init(initialSelection: PostListFilterSettings.AuthorFilter,
+         gravatarEmail: String? = nil,
+         onSelectionChanged: ((PostListFilterSettings.AuthorFilter) -> Void)? = nil) {
+        self.gravatarEmail = gravatarEmail
```

Comment:
Nipticky: can we add a newline between the last signature line, and the first line that actually belongs to the body? (it's a bit hard to grasp, at a glance, where one begins and the other ends)